src/composables/addressBook.ts

```diff
@@ -0,0 +1,234 @@
+import { computed, ref } from 'vue';
+import { Directory, Filesystem } from '@capacitor/filesystem';
+import { Capacitor } from '@capacitor/core';
+
+import {
+  AccountAddress,
+  IAddressBook,
+  IAddressBookEntry,
+  Protocol,
+} from '@/types';
+import { STORAGE_KEYS, MODAL_ADDRESS_BOOK_IMPORT } from '@/constants';
+import { AddressBookEntryExists, AddressBookInvalidAddress, AddressBookRequiredFields } from '@/lib/errors';
+import {
+  convertBlobToBase64,
+  getProtocolByAddress,
+  handleUnknownError,
+  pipe,
+} from '@/utils';
+import { tg as t } from '@/popup/plugins/i18n';
+
+import { createCustomScopedComposable } from './composablesHelpers';
+import { useStorageRef } from './storageRef';
+import { useModals } from './modals';
+import { useFileDialog } from './fileDialog';
+
+interface IAddressBookOptions {
+  name: string;
+  address: AccountAddress;
+  isBookmarked?: boolean;
+}
+
+const addressBook = useStorageRef<IAddressBook>([], STORAGE_KEYS.addressBook);
+
+export const useAddressBook = createCustomScopedComposable(() => {
+  const { openModal, openDefaultModal } = useModals();
+  const { openFileDialog } = useFileDialog();
+
+  const activeFilter = ref<Protocol | null>(null);
```

Comment:
I suggest to rename this to `protocolFilter` (or something similar) as it holds only protocol value.